packages/openapi-generator/src/parser/unique-naming.ts

```diff
@@ -2,107 +2,70 @@ import { UniqueNameGenerator, camelCase } from '@sap-cloud-sdk/util';
 
 /**
  * Ensure uniqueness of names.
- * Takes a list of items, identifies duplicate names and renames the duplicate names.
- * @param items List of items to rename.
- * @param nameHandler Object containing a name getter and/or setter and/or formatter.
- * @param nameHandler.getName Function to get the name of an item. Retrieves the `name` property by default.
- * @param nameHandler.transformItem Function to transform the given item with the new name. Sets the `name` property by default.
- * @param nameHandler.formatName Function to transform the name when finding a unique name. Defaults to camel case.
- * @param nameHandler.reservedWords Reserved words that should be handled as duplicates.
- * @returns The given items with unique names.
+ * Takes a list of names and renames those that are duplicate.
+ * @param names List of names to deduplicate.
+ * @param options Object containing options to configure the transformation.
+ * @param options.format Function to format the name. Defaults to camel case.
+ * @param options.reservedWords Reserved words that should be handled as duplicates.
+ * @returns Unique names in the given order.
  */
-export function ensureUniqueNames<ItemT, UniqueItemT>(
-  items: ItemT[],
-  nameHandler: {
-    getName?: (item: ItemT) => string;
-    transformItem?: (item: ItemT, name: string) => UniqueItemT;
-    formatName?: (name: string) => string;
+export function ensureUniqueNames(
```

Comment:
perhaps the name of the method is not optimal anymore. The methods gives you uniqueNames it does not changes the objects anymore.

Reply:
Also general: Should this go to the util like the unique-name-generator?

Reply:
For now this is not necessary, but once we want to use it somewhere else we should move it out.

Reply:
> perhaps the name of the method is not optimal anymore. The methods gives you uniqueNames it does not changes the objects anymore.

I renamed it when working on it, but `uniqueNames` is a bit annoying because sometimes that is the name of the variable ;)
I was considering `transformUniqueNames` or `formatUniqueNames` but this implies that the names are unique already. I also considered `formatToUniqueNames` but this is rather long, so I stayed with `ensureUniqueNames` for now.
That said, I don't have a strong opinion, let me know what you think, I am already working on the next PR, I would change the name there, then.